core/codegen/src/macros/uri.rs

```diff
@@ -129,6 +129,7 @@ pub fn uri_internal(
     for (i, &(mut ident, ref ty)) in internal.fn_args.iter().enumerate() {
         let (span, mut expr) = (exprs[i].span, exprs[i].clone());
         ident.span = span;
+        let f_ident = Ident::from_str(&("f_".to_string() + &ident.to_string()));
```

Comment:
Can we simply rebind `ident` here so that we don't have two `ident`-like variables? Also, can you add a comment above the redefinition explaining what/why this is happening?